src/echo.js

```diff
@@ -9,26 +11,79 @@ common.register('echo', _echo, {
 //@ Available options:
 //@
 //@ + `-e`: interpret backslash escapes (default)
+//@ + `-n`: remove trailing newline from output
 //@
 //@ Examples:
 //@
 //@ ```javascript
 //@ echo('hello world');
 //@ var str = echo('hello world');
+//@ echo('-n', 'no newline at end');
 //@ ```
 //@
 //@ Prints string to stdout, and returns string with additional utility methods
 //@ like `.to()`.
-function _echo(opts, messages) {
+function _echo(opts) {
   // allow strings starting with '-', see issue #20
-  messages = [].slice.call(arguments, opts ? 0 : 1);
+  var messages = [].slice.call(arguments, opts ? 0 : 1);
+  var options = parseEchoOptions(opts);
+  var output;
 
-  if (messages[0] === '-e') {
-    // ignore -e
+  if (options) {
+    // first argument was options string,
+    // so do not print it
     messages.shift();
+    output = format.apply(null, messages);
+    if (!options.n) {
+      // add newline if -n is not passed
+      output += '\n';
+    }
+  } else {
+    output = format.apply(null, messages) + '\n';
   }
 
-  console.log.apply(console, messages);
-  return messages.join(' ') + '\n';
+  process.stdout.write(output);
+
+  return output;
 }
+
+function parseEchoOptions(opts) {
```

Comment:
Ah, I see the issue. I think it may be better to either:

 - hack around this by temporarily setting `config.silent = true` (document what we're doing in comments)
 - add a way to silence `parseOptions` (exception only, no stdio)

I think the approach we take depends on if we want to expose this feature in `parseOptions` in the plugin API.

@freitagbr What do you think?